models/login_source.go

```diff
@@ -403,6 +403,19 @@ func LoginViaLDAP(user *User, login, password string, source *LoginSource, autoR
 
 	var isAttributeSSHPublicKeySet = len(strings.TrimSpace(source.LDAP().AttributeSSHPublicKey)) > 0
 
+	// Update User if exist
+	isExist, err := IsUserExist(0, sr.Username)
+	if err != nil {
+		return nil, err
+	} else if isExist {
+		if user.ProhibitLogin {
+			return nil, ErrUserProhibitLogin{user.ID, user.Name}
+		} else if len(source.LDAP().AdminFilter) > 0 && user.IsAdmin != sr.IsAdmin {
+			// Change existing admin flag only if AdminFilter option is set
+			go AsyncUpdateUserCols(user, "is_admin")
+		}
+	}
```

Comment:
But now you're not updating the user name at all... ever! 😄

Reply:
@guillep2k -> #1478